src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -105,6 +105,14 @@ function checkAndWarnForMutatedStyle(style1, style2, component) {
   if (shallowEqual(style1, style2)) {
     return;
   }
+  // Don't show mutated style warning if a value is NaN since the styles weren't mutated.
+  for (var key in style1) {
+    var value = style1[key];
+
+    if (isNaN(value) && ('' + value) === 'NaN') {
+      return;
```

Comment:
This will bail out if any of the styles are `NaN`, but I don't think that's what we want.  For instance, suppose one of the styles is `NaN` but the other style was mutated.  In such a case, we want to display the mutation warning for the style that was mutated, and just skip the NaN.